packages/core/android/src/main/java/io/sentry/react/RNSentryMapConverter.java

```diff
@@ -131,4 +136,37 @@ private static void addValueToWritableMap(WritableMap writableMap, String key, O
       logger.log(SentryLevel.ERROR, "Could not convert object" + value);
     }
   }
+
+  public static ReadableMap jsonObjectToReadableMap(JSONObject jsonObject) {
+    Map<String, Object> map = jsonObjectToMap(jsonObject);
+    return mapToReadableMap(map);
+  }
+
+  public static ReadableMap mapToReadableMap(Map<String, Object> map) {
+    // We are not directly using `convertToWritable` since `Arguments.createArray()`
+    // fails before bridge initialisation
+    Object[] keysAndValues = new Object[map.size() * 2];
+    int index = 0;
+    for (Map.Entry<String, Object> entry : map.entrySet()) {
+      keysAndValues[index++] = entry.getKey();
+      keysAndValues[index++] = entry.getValue();
+    }
+    return JavaOnlyMap.of(keysAndValues);
+  }
+
+  private static Map<String, Object> jsonObjectToMap(JSONObject jsonObject) {
+    Map<String, Object> map = new HashMap<>();
+    Iterator<String> keys = jsonObject.keys();
+    while (keys.hasNext()) {
+      String key = keys.next();
+      Object value = null;
+      try {
+        value = jsonObject.get(key);
+      } catch (JSONException e) {
+        throw new RuntimeException(e);
+      }
+      map.put(key, value);
+    }
+    return map;
+  }
```

Comment:
We should check with the Android team, this might be already implemented in SentryAndroid.